zulipterminal/model.py

```diff
@@ -481,6 +481,11 @@ def get_messages(self, *,
         }
         response = self.client.get_messages(message_filters=request)
         if response['result'] == 'success':
+            # Convert message reponse to the modern format
+            response['messages'] = self.modernize_message_response(
+                response['messages']
+            )
```

Comment:
I'm not sure the comment is necessary - it duplicates the function name? :)

Note that we need to call this for all new messages, both on explicit message fetch and for message events.